crates/milli/src/search/facet/filter.rs

```diff
@@ -434,36 +466,56 @@ impl<'a> Filter<'a> {
                 }
             }
             FilterCondition::In { fid, els } => {
-                if crate::is_faceted(fid.value(), filterable_fields) {
-                    let field_ids_map = index.fields_ids_map(rtxn)?;
-                    if let Some(fid) = field_ids_map.id(fid.value()) {
-                        els.iter()
-                            .map(|el| Condition::Equal(el.clone()))
-                            .map(|op| Self::evaluate_operator(rtxn, index, fid, universe, &op))
-                            .union()
-                    } else {
-                        Ok(RoaringBitmap::new())
+                match matching_features(fid.value(), filterable_fields) {
+                    Some(features) if features.is_filterable() => {
+                        let field_ids_map = index.fields_ids_map(rtxn)?;
+                        if let Some(fid) = field_ids_map.id(fid.value()) {
+                            els.iter()
+                                .map(|el| Condition::Equal(el.clone()))
+                                .map(|op| {
+                                    Self::evaluate_operator(
+                                        rtxn, index, fid, universe, &op, &features,
+                                    )
+                                })
+                                .union()
+                        } else {
+                            Ok(RoaringBitmap::new())
+                        }
+                    }
+                    _ => {
+                        let field_ids_map = index.fields_ids_map(rtxn)?;
+                        Err(fid.as_external_error(FilterError::AttributeNotFilterable {
+                            attribute: fid.value(),
+                            filterable_fields: filtered_matching_field_names(
+                                filterable_fields,
+                                &field_ids_map,
+                                &|features| features.is_filterable(),
+                            ),
+                        }))?
                     }
-                } else {
-                    Err(fid.as_external_error(FilterError::AttributeNotFilterable {
-                        attribute: fid.value(),
-                        filterable_fields: filterable_fields.clone(),
-                    }))?
                 }
             }
             FilterCondition::Condition { fid, op } => {
-                if crate::is_faceted(fid.value(), filterable_fields) {
-                    let field_ids_map = index.fields_ids_map(rtxn)?;
-                    if let Some(fid) = field_ids_map.id(fid.value()) {
-                        Self::evaluate_operator(rtxn, index, fid, universe, op)
-                    } else {
-                        Ok(RoaringBitmap::new())
+                match matching_features(fid.value(), filterable_fields) {
```

Comment:
I find it a bit sad that we'll be looking up matching rules each time a fid appears in a condition, but I guess in the common case where a fid appears only once in a filter, it is faster than memorizing the features in a map 🤔

Reply:
I've tried to fix it by relying more on the `FieldIdMapWithMetadata` https://github.com/meilisearch/meilisearch/pull/5254/commits/b88aa9cc76b6af7b3b5f94072b6b63e427fc7b02